src/tree/hist/histogram.h

```diff
@@ -190,6 +193,17 @@ class HistogramBuilder {
           reinterpret_cast<double *>(this->hist_[first_nidx].data()), n);
     }
 
+    if (is_distributed_ && is_col_split_ && is_secure_) {
```

Comment:
Why do we need to allgather the histogram across workers? I thought we only need to send it to the active worker?

Reply:
yes we only need to collect histograms to the active party, but my understanding is we currently do not have a "gather" function to do that?

Reply:
Thank you for sharing, I can look into a gather function in the future.